lib/intermed.ml

```diff
@@ -0,0 +1,32 @@
+module rec TypeDecl : sig
+  module Field : sig
+    type t = { name : string; mutable_ : bool; type_ : Types.type_expr }
+  end
+
+  module Constructor : sig
+    type args = Tuple of Types.type_expr list | Record of Field.t list
+    type t = { name : string; args : args }
+  end
+
+  type param = { type_expr : Types.type_expr }
```

Comment:
This indirection is not necessary atm! I think we'd better remove it as it simplifies the PR since normalize won't have so many modifications.

Reply:
I agree but this won't happen overnight I think so we shouldn't bother with it until then.